Cython/Compiler/PyrexTypes.py

```diff
@@ -4369,6 +4379,117 @@ def error_condition(self, result_code):
         return "dummy"
 
 
+class IndexedPythonType(PyrexType):
```

Comment:
I was considering `PyTypeSubscript` as a better name, but then realised that we are not interested in the subscript property here, but in the fact that we are building a type. So – `PythonTypingTypeConstructor`, maybe?

Reply:
I don't think `PythonTypingTypeConstructor` - in more recent versions of Python you can subscript the types directly and just do `list[int]` (although this PR doesn't handle it). Therefore they not *necessarily* come from the `typing` module. I've changed it to `PythonTypeConstructor` instead.